src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -1,139 +1,132 @@
 
 #include <iostream>
 #include <ctime>
+#include <string.h>
 
 #include "MUSI6106Config.h"
 
 #include "AudioFileIf.h"
+#include "Vibrato.h"
+#include "RingBuffer.h"
 
-using std::cout;
-using std::endl;
+using namespace::std;
 
 // local function declarations
-void    showClInfo ();
+void    showClInfo();
 
 /////////////////////////////////////////////////////////////////////////////////
 // main function
 int main(int argc, char* argv[])
 {
-    std::string             sInputFilePath,                 //!< file paths
-                            sOutputFilePath;
-
-    static const int        kBlockSize = 1024;
-
-    clock_t                 time = 0;
-
-    float                   **ppfAudioData = 0;
-
-    CAudioFileIf            *phAudioFile = 0;
-    std::fstream            hOutputFile;
-    CAudioFileIf::FileSpec_t stFileSpec;
-
+    std::string sInPath;
+    std::string sOutPath;
+    CAudioFileIf    *phInputFile = 0;
+    CAudioFileIf    *phOutputFile = 0;
+    CAudioFileIf::FileSpec_t sFileSpec;
+    CVibrato        *pVibrato = 0;
+    clock_t         time = 0;
+    static const int kBlockSize = 1024;
+    long long iNumFrames = kBlockSize;
+    float fSampleRate;
+    float fModFreq;
+    float fDelay;
+    float fWidth;
+    float **ppfInputData = 0;
+    float **ppfOutputData = 0;
+    
     showClInfo();
-
+    
     //////////////////////////////////////////////////////////////////////////////
     // parse command line arguments
-    if (argc < 2)
-    {
-        cout << "Missing audio input path!";
-        return -1;
+    // arguments should be in this sequence: (input_file_path output_file_path sample_rate mod_frequency mod_delay mod_width)
+    
+    if(argc < 6) {
+        cout << "missing some arguments" << endl;
```

Comment:
This error message is accurate, but it's also not very helpful.
You could consolidate this with the comment above by printing something like "usage: <input_file_path> <output_file_path> ...". This would allow you to remove the comment and help the user out by reminding them of the expected arguments.

Reply:
Thanks, I didn't carefully check the comparing content, I will probably create new branch or update the main branch for comparing for next time.

Reply:
```suggestion
        cout << "missing some arguments" << endl;
        cout << "Usage: <input_path> <sample_rate> <mod_frequency> <delay> <mod_width>" << endl;
        cout << "Example: ./sweep.wav 48000 1000 0.001 0.001" << endl;
```

The input hint now has been printed.